Framework/Algorithms/src/PolarizationCorrections/HeliumAnalyserEfficiency.cpp

```diff
@@ -0,0 +1,493 @@
+  if (!getPropertyValue(PropertyNames::P_HE).empty()) {
+    auto createSingleValuedWorkspace = createChildAlgorithm("CreateSingleValuedWorkspace");
+    createSingleValuedWorkspace->initialize();
+    createSingleValuedWorkspace->setProperty("DataValue", pHe);
+    createSingleValuedWorkspace->setProperty("ErrorValue", pHeError);
+    createSingleValuedWorkspace->setProperty("OutputWorkspace", "phe");
+    createSingleValuedWorkspace->execute();
+    MatrixWorkspace_sptr pheWs = createSingleValuedWorkspace->getProperty("OutputWorkspace");
+    setProperty(PropertyNames::P_HE, pheWs);
+  }
+
+  // Now we have all the parameters to calculate T(lambda), the transmission of the helium
+  // analyser for an incident unpolarised beam. T_para and T_anti are also calculated, the
+  // transmission of the wanted and unwanted spin state. T = (T_para + T_anti) / 2
+
+  if (getPropertyValue(PropertyNames::OUTPUT_T_WORKSPACE).empty() &&
+      getPropertyValue(PropertyNames::OUTPUT_T_PARA_WORKSPACE).empty() &&
+      getPropertyValue(PropertyNames::OUTPUT_T_ANTI_WORKSPACE).empty()) {
+    return;
+  }
+
+  MantidVec tPara, tAnti, tParaErrors, tAntiErrors;
+  calculateTransmission(wavelengthValues, pHe, pHeError, mu, tCrit, tPara, tAnti, tParaErrors, tAntiErrors);
+
+  MatrixWorkspace_sptr tParaWorkspace =
+      createWorkspace("tPara", "Helium Analyser Transmission T_para", wavelengthValues, tPara, tParaErrors);
+  if (!getPropertyValue(PropertyNames::OUTPUT_T_PARA_WORKSPACE).empty()) {
+    setProperty(PropertyNames::OUTPUT_T_PARA_WORKSPACE, tParaWorkspace);
+  }
+
+  MatrixWorkspace_sptr tAntiWorkspace =
+      createWorkspace("tAnti", "Helium Analyser Transmission T_anti", wavelengthValues, tAnti, tAntiErrors);
+  if (!getPropertyValue(PropertyNames::OUTPUT_T_ANTI_WORKSPACE).empty()) {
+    setProperty(PropertyNames::OUTPUT_T_ANTI_WORKSPACE, tAntiWorkspace);
+  }
+
+  if (getPropertyValue(PropertyNames::OUTPUT_T_WORKSPACE).empty()) {
+    return;
+  }
+
+  MatrixWorkspace_sptr transmissionWorkspace = addTwoWorkspaces(tParaWorkspace, tAntiWorkspace);
+
+  auto scale = createChildAlgorithm("Scale");
+  scale->initialize();
+  scale->setProperty("InputWorkspace", transmissionWorkspace);
+  scale->setProperty("OutputWorkspace", transmissionWorkspace);
+  scale->setProperty("Factor", 0.5);
+  scale->setProperty("Operation", "Multiply");
+  scale->execute();
+
+  setProperty(PropertyNames::OUTPUT_T_WORKSPACE, transmissionWorkspace);
+}
+
+void HeliumAnalyserEfficiency::fitAnalyserEfficiency(const double mu, MatrixWorkspace_sptr p, double &pHe,
+                                                     double &pHeError, MantidVec &wavelengthValues, MantidVec &pCalc) {
+  auto fit = createChildAlgorithm("Fit");
+  fit->initialize();
+  fit->setProperty("Function", "name=UserFunction,Formula=tanh(" + std::to_string(mu) + "*phe*x),phe=0.1");
+  fit->setProperty("InputWorkspace", p);
+  const double startLambda = getProperty("StartLambda");
```

Comment:
It would be good to use the property name constants for start and end lambda here, if we can.

Reply:
Done